Added search by email for Stripe Customer ID during member import

sagzy opened the pull request:
refs https://github.com/TryGhost/Product/issues/3593
- when importing members via CSV, it's now possible to add the value "auto" for the "stripe_customer_id" field. When "auto" is passed, the importer will search for a Stripe customer based on the email address provided

sagzy committed:
Added search by email for Stripe Customer ID during member import

refs https://github.com/TryGhost/Product/issues/3593
- when importing members via CSV, it's now possible to add the value "auto" for the "stripe_customer_id" field. When "auto" is passed, the importer will search for a Stripe customer based on the email address provided

SimonBackx commented on ghost/stripe/lib/StripeAPI.js:
```diff
@@ -222,6 +222,30 @@ module.exports = class StripeAPI {
+     */
+    async getCustomerIdByEmail(email) {
+        try {
+            const result = await this._stripe.customers.search({
+                query: `email:"${email}"`,
+                limit: 1
+            });
```
I'm wondering if we shouldn't search more than 1 customer, and return the customer with the newest subscriptions? To avoid reusing an old customer

sagzy replied:
Yes that makes sense, thanks! Updated in [300364f](https://github.com/TryGhost/Ghost/pull/17326/commits/300364f83606933356a91df8025e72b87ee5da41)

sagzy committed:
Updated logic when multiple customers are found

- returns the customer with the most recent subscription, when multiple Stripe customers have the same email address
